core/store/models/job_run.go

```diff
@@ -173,13 +172,14 @@ func (jr *JobRun) ErrorString() string {
 
 // RunRequest stores the fields used to initiate the parent job run.
 type RunRequest struct {
-	ID        uint `gorm:"primary_key"`
-	RequestID *string
-	TxHash    *common.Hash
-	BlockHash *common.Hash
-	Requester *common.Address
-	CreatedAt time.Time
-	Payment   *assets.Link
+	ID            uint `gorm:"primary_key"`
+	RequestID     *string
+	TxHash        *common.Hash
+	BlockHash     *common.Hash
+	Requester     *common.Address
+	CreatedAt     time.Time
+	Payment       *assets.Link
+	RequestParams JSON `gorm:"default: '{}';not null"`
```

Comment:
Why was a default database value chosen here?  I feel this is better set or defined elsewhere.  The database is a strange place to define default values for me.

Reply:
Why is it strange? Honest question.

Reply:
I've never come across this use before.  I've considered the database a place to store and retrieve values, this step seems to want to modify the data too.

Maybe @j16r has some counter points?

Reply:
I don't think it's strange, but I believe this code has no effect unless used with `AutoMigrate`. Line 10 of core/store/migrations/migration1581240419/migrate.go is doing this.

Reply:
Nope, it appears to actually set the default when you save the object.

If I take this line out I get not null violations.

Reply:
This is unexpected behavior for me.  Setting these default values should happen in the codebase, not the database.  If I save a zero value to a database, and it succeeds, I would expect it back.

Reply:
This line sets it in code.

There is _also_ a default in the database, defined separately.

If you intentionally set a null, it will write a null (and explode with not null violation).

Reply:
"There is also a default in the database, defined separately."
Why?

"If you intentionally set a null, it will write a null (and explode with not null violation)."

Isn't this what we want? Othewise why have the 'not null' constraint?